pkg/downloader/chart_downloader.go

```diff
@@ -171,7 +171,11 @@ func (c *ChartDownloader) getOciURI(ref, version string, u *url.URL) (*url.URL,
 		}
 	}
 
-	u.Path = fmt.Sprintf("%s:%s", u.Path, tag)
+	// If the tag is already inside the OCI URI, don't concatenate u.Path with tag.
+	// Otherwise the tag gets invalid.
+	if !(strings.Contains(u.Path, ":"+tag)) {
```

Comment:
I think we should use proper OCI URI parsing, rather than string manipulation. For example (as an extreme edge case), if the first characters of a URI which contains a password match the tag, this will break.

Overall this solution feels like it is working around a larger problem too